src-built-in/components/authentication/app.jsx

```diff
@@ -0,0 +1,84 @@
+/*!
+* The authentication component is a React component that renders a user form to collect auth info and determine whether to give user access to the rest of Finsemble
+* based on the information provided. 
+* Copyright 2020 by ChartIQ, Inc.
+* All rights reserved.
+*/
+
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import { AuthHeader, AuthForm, ErrorMessage, AuthInput, AuthSubmit } from "@chartiq/finsemble-ui/lib/components";
+import useAuth from "@chartiq/finsemble-ui/lib/hooks/useAuth";
+
+export const Authentication = () => {
+    const { saveInputChange, formValues, proceedToAuthorize, quitApplication } = useAuth();
+    const { error, setError } = useState(null);
+
+    /*
+     * Replace this function with your own authentication method that makes the call to your authentication server.
+     */
+    const authenticateUser = async () => {
+        // The user form data to be sent to the authentication server.
+        const { username, password } = formValues;
+        // Replace this with your own authentication calls. Currently it will always authenticate no matter what the user input is.
+        const result = await Promise.resolve({res: "ok"});
+        // the result is an object containing a response and an error. For example:
+        // {
+        //     res: "ok",
+        //     error: "Password is not correct"
+        // }
+        return result;
+    }
+
+    /**
+     * This function invokes the function that makes the call to the server that verifies user authentication information and
+     * decides whether to authorize the user to the rest of the application or display an error on the authentication form.
+     * @param {func} authenticateUser The function to call to authenticate the user.
+     */
+    const onSubmit = async (authenticateUser) => {
+        const authResult = await authenticateUser();
+        if (authResult.res === "ok") {
+            proceedToAuthorize();
+        } else {
+            setError(authResult.error);
+        }
+    };
+
+    return (
+        <>
+            <AuthHeader>
+                <img className="fsbl-logo" />
+                <div className="fsbl-close">
+                    <i className="ff-close" onClick={quitApplication}></i>
+                </div>
+            </AuthHeader>
+
+            <AuthForm>
+                <div className="fsbl-login-logo" alt="Finsemble Sign In"></div>
+                <div className="fsbl-button-wrapper">
+                    {error && <ErrorMessage>
+                        {error}
+                    </ErrorMessage>}
+                    <AuthInput
```

Comment:
I think the auth template should look very small.

```
<AuthUserName/>
<AuthPassword/>
```

What's the justification for these AuthInput components?

Reply:
In case the developers want to add other forms of input, such as lastname, firstname, we will be able to handle it without changing our code. If we're being too specific about the React component we provides, we will need to write additional `<AuthFirstname />` or `<AuthLastname />`, etc.

Reply:
In case the developers want to add other forms of input, such as lastname, firstname, we will be able to handle it without changing our code. If we're being too specific about the React component we provides, we will need to write additional `<AuthFirstname />` or `<AuthLastname />`, etc. This is a typical way of providing React component templates (see Material.UI)